blocks/carousel/carousel.js

```diff
@@ -1,14 +1,19 @@
-import {
-  addEnclosingDiv,
-  createSliderWrapper,
-  setSliderIds,
-  createDottedNav,
-  initSlider,
-  createPictureSlider,
-  createArrowNav,
-} from '../../scripts/lib-carousel.js';
+import Carousel from '../../scripts/lib-carousel.js';
 import { createOptimizedPicture } from '../../scripts/lib-franklin.js';
 
+/**
+ * Add DIV around LI inner text to ensure text is displayed properly
+ * @param block
+ */
+export function addEnclosingDiv(block) {
```

Comment:
if the LI is user-controlled, this may expose us to XSS. instead of using `innerHTML` we should manipulate the DOM:

```
export function addEnclosingDiv(block) {
  const listItems = block.querySelectorAll('.carousel > div > div > ul li');
  listItems.forEach((el) => {
    const div = document.createElement('div');

    // Transfer child nodes from the li to the new div
    while (el.firstChild) {
      div.appendChild(el.firstChild);
    }

    // Append the div back to the li
    el.appendChild(div);
  });
}

```

Reply:
I will remove the function. It was fixed later by karl on highr level and we don't need this anymore.